perun-core/src/main/java/cz/metacentrum/perun/core/implApi/AttributesManagerImplApi.java

```diff
@@ -2601,4 +2601,9 @@
 	 * Copies all values of the attribute to table _attr_u_values which has unique constraint.
 	 */
 	void convertAttributeValuesToUnique(PerunSession session, AttributeDefinition attrDef);
+
+	/**
+	 * Deletes all values of the attribute from table _attr_u_values which has unique constraint.
+	 */
+	void convertAttributeValuesToNonunique(PerunSession session, AttributeDefinition attrDef);
```

Comment:
I am thinking about the way how to check that all expected values were removed. What about changing the behavior of both IMPL and BL methods for conversion to return a number of rows added or deleted? Then this could be at least tested in tests. What do you think?

Reply:
Yes, that sounds nice, I've already tested it manually, because the test already in use basically tests only the change of the attribute to unique/nonunique state... So I've added one more that checks exactly how many rows were changed.